src/Microsoft.Health.Dicom.Core/Features/Workitem/Model/RequirementDetail.cs

```diff
@@ -0,0 +1,26 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using FellowOakDicom;
+using Microsoft.Health.Dicom.Core.Models;
+
+namespace Microsoft.Health.Dicom.Core.Features.Workitem.Model;
+
+public sealed class RequirementDetail
+{
+    public RequirementDetail(DicomTag dicomTag, RequirementCode requirementCode, HashSet<RequirementDetail> sequenceRequirements = default)
```

Comment:
Is there any reason we're making the input type for `sequenceRequirements` instead of `IReadOnlyCollection<RequirementDetail>`?